src/sagas.js

```diff
@@ -304,15 +316,39 @@ function* refreshData (action) {
     console.log('refreshData(): not quiet')
     yield put(loadInProgress({ value: true }))
   }
-  yield put(getAllVms({ shallowFetch: !!action.payload.shallowFetch }))
-  yield put(getAllPools())
+  yield put(getVmsByCount({ count: action.payload.page * AppConfiguration.pageLimit, shallowFetch: !!action.payload.shallowFetch }))
```

Comment:
How does this behave if all pages are already loaded and new VM (or multiple) are added on the server by another user?